programs/clearing_house/src/lib.rs

```diff
@@ -1552,22 +1577,55 @@ pub mod clearing_house {
             )?;
 
             let user_position = &mut user.positions[position_index];
-            user_position.unsettled_pnl = user_position
-                .unsettled_pnl
-                .checked_sub(cast_to_i128(pnl_to_settle)?)
-                .ok_or_else(math_error!())?;
+            controller::position::update_unsettled_pnl(
+                user_position,
+                market,
+                -cast_to_i128(pnl_to_settle)?, // negative sign
+            )?;
         } else {
```

Comment:
we should have some tests for this else block. make sure market pnl gets/gives the right amount based on current total_fees_minus_distributions